### PR TITLE
Upgrade R2DBC implementation classes for spring boot 2.4

### LINKED ISSUE #13613: Upgrade R2DBC implementation for Spring Boot 2.4
##### **Overview of the issue**

From the [Spring Boot 2.4 release notes](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.4-Release-Notes):

> The core infrastructure of R2DBC has moved to Spring Framework with a new `spring-r2dbc` module. If you are using this infrastructure, make sure to migrate deprecated access to the new core support.

I'm creating this as a separate issue than the [main Spring Boot 2.4 issue](https://github.com/jhipster/generator-jhipster/issues/13237) because it requires more work than expected and I want to add a bug bounty to it.

##### **Motivation for or Use Case**

Our current implementation of R2DBC does not work with Spring Boot 2.4. [This PR](https://github.com/jhipster/generator-jhipster/pull/13551) is tracking Spring Boot 2.4 support and apps with R2DBC do not start successfully because the expected `DatabaseClient` is not available.

##### **Reproduce the error**

`npm link` in the `spring-boot-2.4` branch, then create an app with the following `.yo-rc.json` and try to start it.

```json
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "reactive": true,
    "baseName": "sampleWebfluxPsql",
    "packageName": "tech.jhipster.sample",
    "packageFolder": "tech/jhipster/sample",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "buildTool": "maven",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "fr"],
    "testFrameworks": ["gatling", "cypress"],
    "serverPort": "8080",
    "jhiPrefix": "jhi",
    "clientPackageManager": "npm",
    "enableSwaggerCodegen": true,
    "clientFramework": "angularX",
    "creationTimestamp": 1596513172471,
    "jwtSecretKey": "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ=",
    "blueprints": [],
    "otherModules": [],
    "jhipsterVersion": "7.0.0-beta.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "pages": [],
    "serviceDiscoveryType": false,
    "messageBroker": false,
    "clientTheme": "none",
    "clientThemeVariant": "",
    "withAdminUi": true
  }
}
```

##### **Suggest a Fix**

Upgrade to non-deprecated classes.

##### **JHipster Version(s)**

`spring-boot-2.4` branch

### PR DESCRIPTION
**Pull request by ecostanzi**
Fixes #13613 

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by ecostanzi** [comment]
Tests for a project without entities should pass. Currently working on single entity repositories.

**Comment by ecostanzi** [comment]
I still have to clean up imports and some constructor, but webflux sql tests are passing now. I'll continue tomorrow.

**Comment by mraible** [comment]
@ecostanzi It seems that all tests are passing! I think the `npm test` failure is caused by something else. 

You could test some more databases by enabling them in the CI job for this branch.

https://github.com/jhipster/generator-jhipster/blob/main/.github/workflows/webflux.yml#L142

I recommend using `sqllight` for the `entity` property. That way, e2e tests are a bit faster.

**Comment by ecostanzi** [comment]
@mraible I think it's ready for the review. I added the tests as you suggested. I also tested it locally using [blog-oauth2.jdl](https://github.com/jhipster/jdl-samples/blob/main/blog-oauth2.jdl) with the reactive option and it seems to work correctly. Feel free to suggest more tests if they're needed.

**Review by mraible — APPROVED** [review]

**Comment by mraible** [comment]
Well done, @ecostanzi! 🙌

File: .github/workflows/webflux.yml [hunk 1]
```diff
@@ -139,18 +139,18 @@ jobs:
...
-          #  testcontainers: 1
-          #- app-type: webflux-mariadb-gradle
-          #  entity: sqlfull
-          #  profile: prod
-          #  war: 0
-          #  e2e: 1
-          #  testcontainers: 1
+          - app-type: webflux-mysql
+            entity: sqllight
+            profile: prod
```
Comment (mshima) [inline .github/workflows/webflux.yml:144]:
```suggestion
            environment: prod
```

File: .github/workflows/webflux.yml [hunk 2]
```diff
@@ -139,18 +139,18 @@ jobs:
...
+            war: 0
+            e2e: 1
+            testcontainers: 1
+          - app-type: webflux-mariadb-gradle
+            entity: sqllight
+            profile: prod
```
Comment (mshima) [inline .github/workflows/webflux.yml:150]:
```suggestion
            environment: prod
```

**Comment by mshima** [comment]
Not sure why but webflux tests are not been executed.

**Comment by ecostanzi** [comment]
thanks @mshima I didn't noticed :/

**Comment by mraible** [comment]
Please merge when CI passes!

**Comment by ecostanzi** [comment]
With mariadb integration tests with testcontainers don't work, I disabled testcontainers. With mysql integration tests pass, but I noticed some e2e test failures that don't seem to be related to the backend. Have we ever seen them working? I saw that they were both been disabled for similar reasons(see https://github.com/jhipster/generator-jhipster/commit/361fab2d30e92641d42747f9a85e7aa4e90e4186 and https://github.com/jhipster/generator-jhipster/commit/6fcbb1e45cb04ccbffbfddb5dae3340fd8d6dd53).

**Comment by mraible** [comment]
I'm merging. If there are remaining issues, we can clean them up in the `spring-boot-2.4` branch.

**Pull request merged**